cfme/tests/control/test_compliance.py

```
@@ -77,75 +74,52 @@ def assign_policy_for_testing(policy_for_testing, host, policy_profile_name):
-        appl_name = provider.mgmt.get_vm_name_from_ip(ip_addr)
-        appliance = Appliance(provider.key, appl_name)
-        logger.info(
-            "The tested appliance (%s) is already on this provider (%s) so reusing it.",
-            appl_name, provider.key)
-        try:
-            appliance.configure_fleecing()
-        except (EOFError, ApplianceException) as e:
-            # If something was happening, restart and wait for the UI to reappear to prevent errors
-            appliance.ipapp.reboot()
-            pytest.skip(
-                "Error during appliance configuration. Skipping:\n{}: {}".format(
-                    type(e).__name__, str(e)))
-        vm = VM.factory(appl_name, provider)
-    except exceptions.VMNotFoundViaIP:
-        logger.info("Provisioning a new appliance on provider %s.", provider.key)
-        appliance = provision_appliance(
-            vm_name_prefix=PREFIX + "host_",
-            version=str(version.current_version()),
-            provider_name=provider.key)
-        request.addfinalizer(lambda: diaper(appliance.destroy))
-        try:
-            appliance.configure(setup_fleece=True)
-        except (EOFError, ApplianceException) as e:   # Add known exceptions as needed.
-            pytest.skip(
-                "Error during appliance configuration. Skipping:\n{}: {}".format(
-                    type(e).__name__, str(e)))
-        vm = VM.factory(appliance.vm_name, provider)
-    if provider.type in {"rhevm"}:
-        request.addfinalizer(appliance.remove_rhev_direct_lun_disk)
-    # Do the final touches
-    with appliance.ipapp(browser_steal=True) as appl:
-        appl.set_session_timeout(86400)
+        if provider.mgmt.does_vm_exist(vm.name):
+            provider.mgmt.delete_vm(vm.name)
         provider.refresh_provider_relationships()
-        vm.wait_to_appear()
-        vm.load_details()
-        wait_for_ssa_enabled()
-        yield vm
+    except Exception as e:
```

what is it for ? 
you won't see error if there is any. This is bad idea to ignore exceptions even in teardown.